Reconstruction/RecCalorimeter/src/components/MergeCaloHitsTool.cpp

```diff
@@ -0,0 +1,106 @@
+#include "MergeCaloHitsTool.h"
+
+// our EDM
+#include "datamodel/CaloHit.h"
+
+DECLARE_TOOL_FACTORY(MergeCaloHitsTool)
+
+MergeCaloHitsTool::MergeCaloHitsTool(const std::string& type,const std::string& name, const IInterface* parent) 
+  : GaudiTool(type, name, parent)
+{
+  declareInterface<IMergeCaloHitsTool>(this);
+
+}
+
+MergeCaloHitsTool::~MergeCaloHitsTool()
+{
+}
+
+StatusCode MergeCaloHitsTool::initialize() {
+  StatusCode sc = GaudiTool::initialize();
+  if (sc.isFailure()) return sc;
+  return sc;
+
+}
+
+std::vector<fcc::CaloHit*> MergeCaloHitsTool::MergeOneCollection(const fcc::CaloHitCollection& aHits) {
+
+  //Vector of merged hits
+  std::vector<fcc::CaloHit*>  mergedHits;
+
+  //Prepare a map: cellID, index in the vector of cells
+  std::unordered_map<uint64_t, unsigned> cellIdMap;
```

Comment:
As discussed privately, this can be simplified by using `unordered_map<uint64_t, fcc::CaloHit*>`